lib/iris/tests/test_analysis.py

```diff
@@ -1731,188 +1733,145 @@ def setup_test_data(self):
-        weights = iris.analysis._Weights(
-            self.ancillary_variable, self.cube, units="J"
-        )
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, [[5, 6, 7], [5, 6, 7]])
-        assert weights.units == "J"
-
     def test_init_with_cell_measure(self):
-        weights = iris.analysis._Weights(self.cell_measure, self.cube)
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, np.arange(6).reshape(2, 3))
+        weights = _Weights(self.cell_measure, self.cube)
+        assert isinstance(weights.array, self.target_type)
+        assert isinstance(weights.units, cf_units.Unit)
+        np.testing.assert_array_equal(weights.array, self.data)
         assert weights.units == "m2"
 
-    def test_init_with_cell_measure_and_units(self):
-        weights = iris.analysis._Weights(
-            self.cell_measure, self.cube, units="J"
+    def test_get_updated_kwargs_no_weights(self):
+        kwargs = {"test": [1, 2, 3]}
+        (new_kwargs, weights_units) = _Weights.get_updated_kwargs(
+            kwargs, self.cube
         )
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, np.arange(6).reshape(2, 3))
-        assert weights.units == "J"
+        assert new_kwargs is not kwargs
+        assert new_kwargs == {"test": [1, 2, 3]}
+        assert weights_units is None
 
-    def test_init_with_list(self):
-        weights = iris.analysis._Weights([1, 2, 3], self.cube)
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, [1, 2, 3])
-        assert weights.units == "1"
-
-    def test_init_with_list_and_units(self):
-        weights = iris.analysis._Weights([1, 2, 3], self.cube, units="J")
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, [1, 2, 3])
-        assert weights.units == "J"
-
-    def test_init_with_ndarray(self):
-        weights = iris.analysis._Weights(np.zeros((5, 5)), self.cube)
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, np.zeros((5, 5)))
-        assert weights.units == "1"
-
-    def test_init_with_ndarray_and_units(self):
-        weights = iris.analysis._Weights(
-            np.zeros((5, 5)), self.cube, units="J"
+    def test_get_updated_kwargs_weights_none(self):
+        kwargs = {"test": [1, 2, 3], "weights": None}
+        (new_kwargs, weights_units) = _Weights.get_updated_kwargs(
+            kwargs, self.cube
         )
-        assert isinstance(weights, iris.analysis._Weights)
-        np.testing.assert_array_equal(weights, np.zeros((5, 5)))
-        assert weights.units == "J"
+        assert new_kwargs is not kwargs
+        assert new_kwargs == {"test": [1, 2, 3], "weights": None}
+        assert weights_units is None
 
-    def test_init_with_invalid_obj(self):
```

Comment:
I know the circumstances have changed a lot, but shouldn't we still have a test for an object that is of no use? E.g. a `str` or a `list`?

Reply:
Added in https://github.com/SciTools/iris/pull/5341/commits/fb466ba6c1804e79dad81a06f630ddf17494ec79